text/0000-offline-node-improved-trajectories-bags.md

```diff
@@ -0,0 +1,106 @@
+  The ability to do this has to be preserved or expanded.
+
+If we suppose that there is a mandated 1:1 relationship between bags and trajectories in the offline node, the following configuration can be proposed.
+
+### Configuration #1
+
+__For each bag (trajectory)__, the configuration is a __tuple__ consisting of:
+  - `TrajectoryOptions` (given in a `.lua` file)
+  - A set of sensor data topics
+  - Boolean `use_bag_transforms` which allows us to ignore transforms in the bag
+  - An optional `.urdf` file.
+
+Both `TrajectoryOptions` and the sensor data topics can be identical for multiple different bags (preserving __B2c__). However, in addition to that, the trajectories can be concurrent (__B3c__).
+
+Because the trajectories may be concurrent, due care would need to be taken to ensure that data published on the same topic, but in different bags, gets routed to the appropriate trajectory.
+Also, each trajectory should have its own transform buffer, to avoid interference between different bags.
+  - For example, we may have two identical separate robots driving concurrently (__B3__); all the frames and topics are identical for both robots.
+  The online node cannot handle this (we would have to ensure that the sensor data topics would be different by e.g. using different namespaces for each robot, as well as having different frames; conflicting transforms with same frames would be an unsolvable nightmare). 
+  - However, if we had a separate bag for each robot (__B3c__), the offline node would have the advantage of knowing which data originated from which bag, so it could handle this case.
+
+Going further, allowing multiple trajectories in a single bag (__C 2/3 a__) and relaxing the 1 trajectory : 1 bag restriction would mean that the configuration of each bag becomes a set of tuples mentioned above.
+This is given in Configuration #2.
+
+### Configuration #2
+
+Similar to Configuration #1, but have a __set of tuples for each bag__.
+Each tuple describes a trajectory in the bag.
+
+For both Configuration #1 and Configuration #2, all concurrent trajectories (whether in the same bag as permitted by Configuration #2, or in different bags) are __collated__.
+
+### Handling split bags
+
+Another common use case is (__b__) - a single continuous trajectory is split into multiple bags (e.g. when using the split option of `rosbag record`).
+
+This could be handled by defining a `BagAggregate`: a list of one or more continuous bags, which are to be treated as one bag.
+
+### Configuration #3
+Configuration is given as a __set of tuples for each `BagAggregate`__, each tuple describing a trajectory in the `BagAggregate`.
+
+As before, any concurrent trajectories are collated.
+Each `BagAggregate` has its own transform buffer (which is shared by all bags in the aggregate).
+
+### Data format of the configuration
```

Comment:
This section is most interesting to me - but it needs to be written. For example would we replace the current command line options or add a new one that reads a lua file as input? How would such a Lua file look?

Reply:
#helpneeded :D I am currently a bit short on time, but I'll get back to this RFC when I can.